pandas/core/dtypes/cast.py

```diff
@@ -1018,6 +1019,80 @@ def soft_convert_objects(
     return values
 
 
+def convert_dtypes(
+    input_array,
```

Comment:
can we type this Union[EA, PA] (do we have a type for this)?

Reply:
can you type the return: Dtype ?

Reply:
> can we type this Union[EA, PA] (do we have a type for this)?
Currently no type for `PandasArray`, so I'm a bit hesitant to muck around with `_typing.py` to try to get that to work.

> can you type the return: Dtype ?
Done